modules/generator/registry/counter.go

```diff
@@ -23,6 +21,7 @@ type counter struct {
 }
 
 type counterSeries struct {
+	labels []string
```

Comment:
Nit: in Prometheus, the key-value pair is as a whole called `label`. We could use that terminology and simplify the two slices into one.